[2024-09-12T13:49:27.557+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-12T13:49:27.621+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_find_makers_dag.spark_submit_task manual__2024-09-12T13:49:20.694525+00:00 [queued]>
[2024-09-12T13:49:27.644+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_find_makers_dag.spark_submit_task manual__2024-09-12T13:49:20.694525+00:00 [queued]>
[2024-09-12T13:49:27.645+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 4
[2024-09-12T13:49:27.680+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark_submit_task> on 2024-09-12 13:49:20.694525+00:00
[2024-09-12T13:49:27.694+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=114) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-12T13:49:27.696+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_find_makers_dag', 'spark_submit_task', 'manual__2024-09-12T13:49:20.694525+00:00', '--job-id', '330', '--raw', '--subdir', 'DAGS_FOLDER/spark_sql_dag.py', '--cfg-path', '/tmp/tmptnu7r74t']
[2024-09-12T13:49:27.697+0000] {standard_task_runner.py:72} INFO - Started process 115 to run task
[2024-09-12T13:49:27.698+0000] {standard_task_runner.py:105} INFO - Job 330: Subtask spark_submit_task
[2024-09-12T13:49:27.823+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_find_makers_dag.spark_submit_task manual__2024-09-12T13:49:20.694525+00:00 [running]> on host c2cd4232b716
[2024-09-12T13:49:28.127+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Denis' AIRFLOW_CTX_DAG_ID='spark_find_makers_dag' AIRFLOW_CTX_TASK_ID='spark_submit_task' AIRFLOW_CTX_EXECUTION_DATE='2024-09-12T13:49:20.694525+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-12T13:49:20.694525+00:00'
[2024-09-12T13:49:28.133+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-12T13:49:28.252+0000] {spark_submit.py:273} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-09-12T13:49:28.257+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=local[*] --name arrow-spark --verbose /opt/***/dags/spark_job.py
[2024-09-12T13:49:28.728+0000] {spark_submit.py:579} INFO - JAVA_HOME is not set
[2024-09-12T13:49:28.753+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=local[*] --name arrow-spark --verbose /opt/***/dags/spark_job.py. Error code is: 1.
[2024-09-12T13:49:28.781+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_find_makers_dag, task_id=spark_submit_task, run_id=manual__2024-09-12T13:49:20.694525+00:00, execution_date=20240912T134920, start_date=20240912T134927, end_date=20240912T134928
[2024-09-12T13:49:28.840+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-12T13:49:28.842+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 330 for task spark_submit_task (Cannot execute: spark-submit --master yarn --conf spark.master=local[*] --name arrow-spark --verbose /opt/***/dags/spark_job.py. Error code is: 1.; 115)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=local[*] --name arrow-spark --verbose /opt/***/dags/spark_job.py. Error code is: 1.
[2024-09-12T13:49:28.906+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-12T13:49:28.953+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-12T13:49:28.960+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
